src/inmanta/plugins.py

```diff
@@ -306,27 +313,26 @@ def to_dsl_type(python_type: type[object]) -> inmanta_type.Type:
                 args = typing.get_args(python_type)
                 if not args:
                     return inmanta_type.List()
-                return inmanta_type.TypedList(to_dsl_type(args[0]))
+                return inmanta_type.TypedList(to_dsl_type(args[0], location, resolver))
             else:
                 raise TypingException(None, f"invalid type {python_type}, list types should be Sequence or list")
 
         # Set
         if issubclass(origin, collections.abc.Set):
             raise TypingException(None, f"invalid type {python_type}, set is not supported on the plugin boundary")
 
-    # TODO annotated types
-    # if typing.get_origin(t) is typing.Annotated:
-    #     args: Sequence[object] = typing.get_args(python_type)
-    #     inmanta_types: Sequence[plugin_typing.InmantaType] =
-    #     [arg if isinstance(arg, plugin_typing.InmantaType) for arg in args]
-    #     if inmanta_types:
-    #         if len(inmanta_types) > 1:
-    #             # TODO
-    #             raise Exception()
-    #         # TODO
-    #         return parse_dsl_type(inmanta_types[0].dsl_type)
-    #     # the annotation doesn't concern us => use base type
-    #     return to_dsl_type(args[0])
+    if typing.get_origin(python_type) is typing.Annotated:
+        annotated_args = typing.get_args(python_type)
+        inmanta_types: Sequence[plugin_typing.InmantaType] = [
+            arg for arg in annotated_args if isinstance(arg, plugin_typing.InmantaType)
+        ]
+        if inmanta_types:
+            if len(inmanta_types) > 1:
+                raise TypingException(None, f"invalid type {python_type}, only one InmantaType annotation is supported")
+            return parse_dsl_type(inmanta_types[0].dsl_type, location, resolver)
+        # the annotation doesn't concern us => use base type
+        return to_dsl_type(annotated_args[0], location, resolver)
```

Comment:
> @sanderr What happens if we can't find the InmantaType? Do we default to Any?

If it's not there, it simply means that the `Annotated` has nothing to do with us. e.g. `typing.Annotated[str, ContextForSomeMypyPlugin()]`. So as far as our type-checking is concerned, it's just `str`. That's why we fall back to the base type here.

Reply:
I don't think that is the case.
If I have `Annotated[str, "something"]` it will default to string.
But if I have `Annotated[str, plugin_typing.InmantaType("unexistent-entity")]` It should fail on `parse_dsl_type`.

Reply:
Oh, I misunderstood the question. I'd make sure to treat it the same as any other DSL-type plugin annotation. I would expect that means to raise an exception. i.e. what if you do `def my_plugin(arg: "not-a-valid-type") -> None: ...`?